e2e/integration_tests/data_flow_test.go

```diff
@@ -242,6 +245,123 @@ func Test_EndpointLinksAreAvailable(t *testing.T) {
+	th.Assert(t, err == nil, err)
+	th.Assert(t, hitp.CHPLID == testHITP.CHPLID, "CHPL ID is not what was expected")
+	th.Assert(t, hitp.CertificationEdition == testHITP.CertificationEdition, "Certification edition is not what was expected")
+	th.Assert(t, hitp.Developer == testHITP.Developer, "Developer is not what was expected")
+	th.Assert(t, hitp.CertificationDate.Equal(testHITP.CertificationDate), "Certification date is not what was expected")
+	th.Assert(t, hitp.CertificationStatus == testHITP.CertificationStatus, "Certification status is not what was expected")
+	th.Assert(t, reflect.DeepEqual(hitp.CertificationCriteria, testHITP.CertificationCriteria), "Certification criteria is not what was expected")
+}
+
+func Test_RetrieveCapabilityStatements(t *testing.T) {
+	var err error
+	qUser := viper.GetString("quser")
+	qPassword := viper.GetString("qpassword")
+	qHost := viper.GetString("qhost")
+	qPort := viper.GetString("qport")
+	qName := viper.GetString("qname")
+
+	hap := th.HostAndPort{Host: qHost, Port: qPort}
+	err = th.CheckResources(hap)
+	if err != nil {
+		panic(err)
+	}
+
+	var mq lanternmq.MessageQueue
+	var chID lanternmq.ChannelID
+	mq, chID, err = aq.ConnectToServerAndQueue(qUser, qPassword, qHost, qPort, qName)
+	defer mq.Close()
+	th.Assert(t, err == nil, err)
+	th.Assert(t, mq != nil, "expected message queue to be created")
+	th.Assert(t, chID != nil, "expected channel ID to be created")
+
+	ctx, cancel := context.WithCancel(context.Background())
+	go capabilityhandler.ReceiveCapabilityStatements(ctx, store, mq, chID, qName)
+	time.Sleep(30 * time.Second)
+	cancel()
+
+	query_str := store.DB.QueryRow("SELECT COUNT(*) FROM fhir_endpoints_info where capability_statement is not null;")
+	var capability_statement_count int
+	err = query_str.Scan(&capability_statement_count)
+	failOnError(err)
+	if capability_statement_count == 0 {
+		t.Fatalf("Fhir_endpoints_info db should have capability statements")
+	}
+
+	query_str = store.DB.QueryRow("SELECT COUNT(capability_statement->>'fhirVersion') FROM fhir_endpoints_info;")
+	var fhir_version_count int
+	err = query_str.Scan(&fhir_version_count)
+	failOnError(err)
+	if fhir_version_count < 300 {
+		t.Fatalf("There should be at least 300 capability statement with fhir version specified")
+	}
+
+	expected_vendor_list := [5]string{"Epic Systems Corporation", "Allscripts", "Medical Information Technology, Inc. (MEDITECH)", "Cerner Corporation", "CareEvolution, Inc."}
+	rows, err := store.DB.Query("SELECT DISTINCT vendor FROM fhir_endpoints_info where vendor!='';")
+	failOnError(err)
+	var test_vendor_list []string
+	defer rows.Close()
+	for rows.Next() {
+		var vendor string
+		err = rows.Scan(&vendor)
+		failOnError(err)
+		test_vendor_list = append(test_vendor_list, vendor)
+	}
+	th.Assert(t, len(expected_vendor_list) == len(test_vendor_list), "Number of distinct vendors is not what was expected")
```

Comment:
i occasionally get this error:

![image](https://user-images.githubusercontent.com/11574199/80770727-04dc1280-8b1f-11ea-9640-6de2f06836cb.png)

this relies on actual responses and the timing of adding things to the queue, etc., so it's possible (at appears happens) that an endpoint with a particular vendor won't be received in time. perhaps just confirm that the most common vendor exists (Epic) (maybe two most common?) is in the list. This will show that the vendor mapping is being executed.